### ISSUE TITLE
Border on localpca and mp-pca

### ISSUE THREAD

**Issue by skoudoro** [nipy/dipy#1945]
Just a reminder for fixing [this comment](https://github.com/nipy/dipy/pull/1917#issuecomment-517330077)

**Comment by RafaelNH** [comment]
Yes - I will give a look to this issue! Thanks @skoudoro, thanks @15thai!

**Comment by RafaelNH** [comment]
@skoudoro, @15thai - I just give a look to this issue but I didn't detect any problem. The centre of the sliding window goes from ``patch_radius`` to ``arr.shape - patch_radius`` but that doesn't mean that edges are not being corrected. Please note that we use all the voxels of the sliding windows for PCA denoising exactly to reach the edges. Overlapping voxels by the sliding window are then combined using the overcomplete average.

If we had an edge issue, the first test on test_lpca.py would fail - test test_lpca_static() - unless there is some special case that I am not seeing. Please let me know!

**Comment by Garyfallidis** [comment]
@RafaelNH The bias will come from the overcomplete averaging of the patches. The corner voxels will not be updated as much as their neighboring voxels.

**Comment by RafaelNH** [comment]
Indeed, corner voxels will not be updated as much, but we take that in to account on the weights of the average. One might argue that this might create a precision issue but not a bias.

Anyway, one might think on many strategies to fill corner voxels (overcomplete averaging was the one we implemented from the work of Manjon); however, this should not be classified as a bug on the code.

**Comment by Garyfallidis** [comment]
This is affecting all boundary voxels (not only corners). The fix should be trivial using np.pad.
However, I do agree that this is not a critical issue. Still it should be addressed with some data that have slices on the boundaries. Most data will just have black background on the borders. But not all.

**Comment by skoudoro** [comment]
> this should not be classified as a bug on the code

Tag updated 😄. Indeed, it is not a critical issue, it is, for this reason, it can wait for the next release
